vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh

```diff
@@ -83,13 +83,18 @@ echo "Granting access to $disk_resource_id for 1 hour"
 # shellcheck disable=SC2102
 sas=$(az disk grant-access --ids $disk_resource_id --duration-in-seconds 3600 --query [accessSas] -o tsv)
 
-# TBD: Need to investigate why `azcopy-preview login --login-type=MSI` does not work
-echo "Setting azcopy environment variables with pool identity: $AZURE_MSI_RESOURCE_STRING"
-export AZCOPY_AUTO_LOGIN_TYPE="MSI"
-export AZCOPY_MSI_RESOURCE_STRING="$AZURE_MSI_RESOURCE_STRING"
-
 echo "Uploading $disk_resource_id to ${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd"
-azcopy-preview copy "${sas}" "${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd" --recursive=true || exit $?
+
+if [[ "${OS_TYPE}" == "Linux" ]]; then
+  azcopy login --login-type=MSI
```

Comment:
Do we know which MSI is attached to?  Are we in the agent at this time ?

Reply:
I believe this references the VMs system assigned identity, as all the docs show AzCopy requires either a Client/Object ID or MSI resource string to use a user-assigned identity. @djsly

Reply:
this is a user-assigned identity, you only need to specify a client/object/resource ID when using a UAMI if there are multiple identities attached to the same VM

Reply:
in this case it's the identity we attach to the pool agents

Reply:
Thanks for the explanation!